internal/pkg/cli/run_local.go

```diff
@@ -612,6 +630,50 @@ func sessionEnvVars(ctx context.Context, sess *session.Session) (map[string]stri
 	return env, nil
 }
 
+func (o *runLocalOpts) taskRoleCredentials(ctx context.Context) (map[string]string, error) {
+	// assumeRoleMethod tries to directly call sts:AssumeRole for TaskRole using default session
+	assumeRoleMethod := func() (map[string]string, error) {
+		taskDef, err := o.ecsClient.TaskDefinition(o.appName, o.envName, o.wkldName)
+		if err != nil {
+			return nil, err
+		}
+
+		taskRoleSess, err := o.sessProvider.FromRole(aws.StringValue(taskDef.TaskRoleArn), o.targetEnv.Region)
+		if err != nil {
+			return nil, err
+		}
+
+		creds, err := sessionEnvVars(ctx, taskRoleSess)
+		if err != nil {
+			return nil, err
+		}
+
+		return creds, nil
+	}
+
+	// ecsExecMethod tries to use ECS Exec to retrive credentials from running container
+	ecsExecMethod := func() (map[string]string, error) {
+		return nil, errors.New("ecs exec method not implemented")
+	}
+
+	credentialsChain := []func() (map[string]string, error){
```

Comment:
It might be over-generic right now and we can do this when we impl ecs exec as a fallback method? The error msg chain doesn't make a lot of sense to users as well when they see "ecs exec method not implemented" 🤔